kpatch-build/create-diff-object.c

```diff
@@ -608,16 +665,40 @@ static void kpatch_compare_sections(struct list_head *seclist)
 
 	/* sync symbol status */
 	list_for_each_entry(sec, seclist, list) {
-		if (is_rela_section(sec)) {
-			if (sec->base->sym && sec->base->sym->status != CHANGED)
-				sec->base->sym->status = sec->status;
-		} else {
-			if (sec->sym && sec->sym->status != CHANGED)
-				sec->sym->status = sec->status;
+		struct symbol *sym;
+		if (is_rela_section(sec))
+			sym = sec->base->sym;
+		else
+			sym = sec->sym;
```

Comment:
optional: could probably do `sym = is_rela_section(sec) ? sec->base->sym : sec->sym;` which I think would be more succinct and readable